src/Gui/OverlayWidgets.h

```diff
@@ -0,0 +1,516 @@
+ *                                                                          *
+ *   This library is free software; you can redistribute it and/or          *
+ *   modify it under the terms of the GNU Library General Public            *
+ *   License as published by the Free Software Foundation; either           *
+ *   version 2 of the License, or (at your option) any later version.       *
+ *                                                                          *
+ *   This library  is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of         *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          *
+ *   GNU Library General Public License for more details.                   *
+ *                                                                          *
+ *   You should have received a copy of the GNU Library General Public      *
+ *   License along with this library; see the file COPYING.LIB. If not,     *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,          *
+ *   Suite 330, Boston, MA  02111-1307, USA                                 *
+ *                                                                          *
+ ****************************************************************************/
+
+#ifndef FC_OVERLAYWIDGETS_H
+#define FC_OVERLAYWIDGETS_H 
+
+#include <QTabWidget>
+#include <QTimer>
+#include <QTime>
+#include <QAction>
+#include <QToolButton>
+#include <QGraphicsEffect>
+#include <QImage>
+#include <QDockWidget>
+#include <QSplitter>
+#include <QMenu>
+
+#include <Base/Parameter.h>
+
+class QPropertyAnimation;
+class QLayoutItem;
+
+namespace Gui {
+
+class OverlayTabWidget;
+
+class GuiExport OverlayManager : public QObject {
+    Q_OBJECT
+public:
+    OverlayManager();
+    virtual ~OverlayManager();
+
+    void restore();
+    void save();
+    void retranslate();
+    void refreshIcons();
+
+    enum ReloadMode {
+        ReloadPending = 0,
+        ReloadPause = 1,
+        ReloadResume = 2,
+    };
+    void reload(ReloadMode mode = ReloadPending);
+
+    void refresh(QWidget *widget=nullptr, bool refreshStyle=false);
+
+    void setupTitleBar(QDockWidget *);
+
+    enum OverlayMode {
+        ToggleActive,
+        ToggleTransparent,
+        EnableActive,
+        DisableActive,
+        EnableAll,
+        DisableAll,
+        ToggleAll,
+        TransparentAll,
+        TransparentNone,
+        ToggleTransparentAll,
+    };
+    void setOverlayMode(OverlayMode mode);
+
+    void changeOverlaySize(int changes);
+
+    void setMouseTransparent(bool enabled);
+    bool isMouseTransparent() const;
+
+    bool isUnderOverlay() const;
+
+    bool isChangingMode() const;
+
+    void initDockWidget(QDockWidget *, QWidget *);
+    void setupDockWidget(QDockWidget *, int dockArea = Qt::NoDockWidgetArea);
+    void unsetupDockWidget(QDockWidget *);
```

Comment:
Without any comments it's hard to tell what these actually do -- what is the difference between "init" and "setup"? Overall I think this file, and the individual classes in it, are too long, too complex, and lack documentation. The first two problems we can work on over time, but only if there is documentation to get started.